src/main/java/io/aiven/kafka/connect/http/sender/AbstractHttpSender.java

```diff
@@ -48,12 +48,25 @@ protected AbstractHttpSender(
     }
 
     public final HttpResponse<String> send(final String body) {
-        final var requestBuilder =
-                httpRequestBuilder.build(config).POST(HttpRequest.BodyPublishers.ofString(body));
+        final var requestBuilder = prepareRequest(body);
         return sendWithRetries(requestBuilder, HttpResponseHandler.ON_HTTP_ERROR_RESPONSE_HANDLER,
                 config.maxRetries());
     }
 
+    // seth http bethod based on config
+    private Builder prepareRequest(final String body) {
+        switch (config.httpMethod()) {
+            case POST:
+                return httpRequestBuilder
+                        .build(config).POST(HttpRequest.BodyPublishers.ofString(body));
+            case PUT:
+                return httpRequestBuilder
+                        .build(config).PUT(HttpRequest.BodyPublishers.ofString(body));
```

Comment:
let's move the repeated code up
```suggestion
        final var httpRequest = httpRequestBuilder.build(config);
        switch (config.httpMethod()) {
            case POST:
                return httpRequest.POST(HttpRequest.BodyPublishers.ofString(body));
            case PUT:
                return httpRequest.PUT(HttpRequest.BodyPublishers.ofString(body));
```